internal/databases/postgres/daemon_handler.go

```diff
@@ -0,0 +1,132 @@
+package postgres
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"net"
+	"os"
+	"path"
+
+	"github.com/wal-g/tracelog"
+)
+
+type CheckType byte
+type FileType byte
+
+type MessageSocketHandler interface {
+	Handle(messageBody []byte, c net.Conn, f func(string) error) error
+}
+
+func MessageTypeConstruct(messageType byte) MessageSocketHandler {
+	switch messageType {
+	case 'C':
+		return CheckType('C')
+	case 'F':
+		return FileType('F')
+	default:
+		return nil
+	}
+}
+
+func (msg CheckType) Handle(messageBody []byte, c net.Conn, f func(string) error) error {
+	_, err := c.Write([]byte{'O', 0, 3})
+	if err != nil {
+		tracelog.ErrorLogger.Printf("Error on writing in socket: %v", err)
+		return err
+	}
+	return nil
+}
+
+func (msg FileType) Handle(messageBody []byte, c net.Conn, f func(string) error) error {
```

Comment:
Why do you need to inject this handler function so far?

Just call the archiver command directly from here:
```go
fullPath := path.Join(pathToWal, walFileName)
tracelog.InfoLogger.Printf("starting wal-push for %s\n", fullPath)
return HandleWALPush(uploader, fullPath)
```